app/models/spree/gateway/braintree_gateway_decorator.rb

```diff
@@ -45,10 +47,14 @@ def authorize money, credit_card, options={}
       # Include device data if present
       options[:device_data] = credit_card.device_data
 
+      current_merchant_account_id = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
+      options[:merchant_account_id] = current_merchant_account_id
+      credit_card.payment_method.preferences[:merchant_account_id] = current_merchant_account_id
+
       if (money.to_f / 100) >= credit_card.payment_method.preferred_three_d_threshold
         # Is expensive enough that three_d_secure should be used
-        options[:payment_method_nonce] = true
-        options[:three_d_secure] = {required: true}
+        options[:payment_method_nonce] = credit_card.encrypted_data
+        options[:three_d_secure] = { required: true }
```

Comment:
Actually did just hit a reason this should be `true` instead of `credit_card.encrypted_data`. If 3D secure is required but the client fails to send the param for some reason then `credit_card.encrypted_data` will be `nil`. This will cause it to [take a different code path](https://github.com/eric1234/active_merchant/blob/braintree-cse-1.47.0/lib/active_merchant/billing/gateways/braintree_blue.rb#L535).

Either way an error will result but by taking the wrong code path the error is less obvious making the root issue harder to track down. I found this out because the mobile folks were not providing the 3D secure token. Took me a bit to figure out the reason because of the unexpected code path.